flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/ScalarOperatorGens.scala

```diff
@@ -2161,6 +2162,7 @@ object ScalarOperatorGens {
 
   def toCodegenCastContext(ctx: CodeGeneratorContext): CodeGeneratorCastRule.Context = {
     new CodeGeneratorCastRule.Context {
+      override def legacyBehaviour(): Boolean = ctx.addReusableLegacyCastBehaviour()
```

Comment:
Can't we just return `true` or `false` here? You can use `primitiveLiteralForType` to convert `ctx.tableConfig.getLegacyCastBehaviour` to string

Reply:
I just thought it's nicer to have the value passed from the ExecutionConfigOption, so that we can easily flip the value, instead of have a hardcoded, true (or false) here.